src/app/pages/pass-finalisation/cat-a-mod2/pass-finalisation.cat-a-mod2.page.ts

```diff
@@ -1,29 +1,118 @@
+import { AuthenticationProvider } from '@providers/authentication/authentication';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { StoreModel } from '@shared/models/store.model';
+import { OutcomeBehaviourMapProvider } from '@providers/outcome-behaviour-map/outcome-behaviour-map';
+import { FormGroup } from '@angular/forms';
+import { Observable, Subscription, merge } from 'rxjs';
+import { GearboxCategory } from '@dvsa/mes-test-schema/categories/common';
+import { ActivityCodes } from '@shared/models/activity-codes';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { map } from 'rxjs/operators';
+import { PersistTests } from '@store/tests/tests.actions';
+import {
+  PASS_CERTIFICATE_NUMBER_CTRL,
+} from '@pages/pass-finalisation/components/pass-certificate-number/pass-certificate-number.constants';
+import {
+  PassFinalisationValidationError,
+  PassFinalisationViewDidEnter,
+} from '@pages/pass-finalisation/pass-finalisation.actions';
 import { TestFlowPageNames } from '@pages/page-names.constants';
+import { TransmissionType } from '@shared/models/transmission-type';
+import { behaviourMap } from '../../office/office-behaviour-map.cat-a-mod2';
+
+interface CatAMod2PassFinalisationPageState {}
+
+type PassFinalisationPageState = CommonPassFinalisationPageState & CatAMod2PassFinalisationPageState;
 
 @Component({
   selector: 'app-pass-finalisation-cat-a-mod2',
   templateUrl: './pass-finalisation.cat-a-mod2.page.html',
-  styleUrls: ['./pass-finalisation.cat-a-mod2.page.scss'],
+  styleUrls: ['./../pass-finalisation.page.scss'],
 })
-export class PassFinalisationCatAMod2Page implements OnInit {
+export class PassFinalisationCatAMod2Page extends PassFinalisationPageComponent implements OnInit {
+
+  pageState: PassFinalisationPageState;
+  form: FormGroup;
+  testOutcome: string = ActivityCodes.PASS;
+  merged$: Observable<string | boolean>;
+  transmission: GearboxCategory;
+  subscription: Subscription;
+  testCategory: TestCategory;
 
   constructor(
-    private navController: NavController,
+    platform: Platform,
+    authenticationProvider: AuthenticationProvider,
+    router: Router,
+    store$: Store<StoreModel>,
     public routeByCat: RouteByCategoryProvider,
-  ) { }
+    private outcomeBehaviourProvider: OutcomeBehaviourMapProvider,
+  ) {
+    super(platform, authenticationProvider, router, store$);
+    this.form = new FormGroup({});
+    this.outcomeBehaviourProvider.setBehaviourMap(behaviourMap);
+  }
+
+  ngOnInit(): void {
+    super.onInitialisation();
+
+    this.pageState = {
+      ...this.commonPageState,
+    };
+
+    const {
+      transmission$, testCategory$,
+    } = this.pageState;
 
-  ngOnInit() {
+    this.merged$ = merge(
+      transmission$.pipe(map((value) => this.transmission = value)),
+      testCategory$.pipe(map((value) => this.testCategory = value as TestCategory)),
```

Comment:
Do we need ```testCategory$```? If not, we can also remove ```testCategory``` on line 48

Reply:
looks like it's passed into <bike-category-type> component in the html

Reply:
```testCategory$``` exists in the base page, why would we not just do ```pageState.testCategory$ | async" in the template?